controller/controller.go

```diff
@@ -585,7 +585,7 @@ func NewSkipKey(key string) error {
 	return skipKeyError{key: key}
 }
 
-// permanentError is an error that is considered not transient.
+// skipKeyError is an error that is indicating that a key was skipped.
```

Comment:
super-nit, but since we're editing this line anyway :)

```suggestion
// skipKeyError is an error that indicates a key was skipped.
```